R/parallel.R

```diff
@@ -224,3 +224,22 @@ hipercow_parallel_setup_parallel <- function(processes, cores_per_process) {
 print.hipercow_parallel <- function(x, ...) {
   print_simple_s3(x, "hipercow parallel control (hipercow_parallel)")
 }
+
+parallel_validate <- function(parallel, cores) {
+  if (!is.null(parallel) && !is.null(parallel$method) &&
+      cores == 1) {
+    cli::cli_abort(c(
+      "You chose parallel method '{parallel$method}', with 1 core",
+      "You need multiple cores for this - check your hipercow_resources"))
+  }
+
+  if (!is.null(parallel) && !is.null(parallel$cores_per_process) &&
+      parallel$cores_per_process > cores) {
+    cli::cli_abort(c(
+      paste("You chose {parallel$cores_per_process} core{?s} per process,",
+            "but requested only {cores} core{?s} in total")))
+
```

Comment:
```suggestion
```

Lots of these blank lines creep in, I tend to drop them when I next edit the file - I've not yet worked out if there's a pattern as to where you put them though...

Reply:
It's a bit subjective! Where I find a block of code starts to get hard to read, and there's a natural pause for breath...!

Reply:
I made this function better, and it didn't need the blank line to look clear to me anymore...!